src/nbl/asset/material_compiler/CMaterialCompilerGLSLBackendCommon.cpp

```diff
@@ -1184,8 +1200,23 @@ auto CMaterialCompilerGLSLBackendCommon::compile(SContext* _ctx, IR* _ir, E_GENE
 	res.usedRegisterCount = 0u;
 	res.globalPrefetchRegCountFlags = 0u;
 
-	for (const IR::INode* root : _ir->roots)
-	{
+	core::unordered_map<IR::CEmitterNode*, uint32_t> emitterCache;
```

Comment:
you're hashing pointers to emitter nodes, not their actual values, I'd be surprised if this cache worked at all

Reply:
anyway since the emitter is supposed to be a root node with the whole BSDF as the child, can it be cached at all?